geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/index/HBaseZ2Index.scala

```diff
@@ -24,18 +24,20 @@ case object HBaseZ2Index extends HBaseLikeZ2Index with HBasePlatform {
                                                sft: SimpleFeatureType,
                                                filter: HBaseFilterStrategyType,
                                                transform: Option[(String, SimpleFeatureType)]): Seq[HFilter] = {
+    import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
     val z2Filter = Z2Index.currentProcessingValues.map { case Z2ProcessingValues(_, bounds) =>
-      configureZ2PushDown(bounds)
+      val offset = if (sft.isTableSharing) { 2 } else { 1 } // sharing + shard
```

Comment:
Blergh.  Having the logic without it working is a confusion to us.  That said, this'll make it *slightly* easier to enable later.  I think it is ok as-is.